python/cudf/cudf/api/types.py

```diff
@@ -154,10 +154,8 @@ def _is_scalar_or_zero_d_array(val):
         Return True if given object is scalar.
     """
     return (
-        (isinstance(val, (np.ndarray, cp.ndarray)) and val.ndim == 0)
-        or (isinstance(val, pd.Categorical) and len(val) == 1)
-        or is_scalar(val)
-    )
+        isinstance(val, (np.ndarray, cp.ndarray)) and val.ndim == 0
```

Comment:
How/why are we able to discard the `Categorical` check?

Reply:
Why? Because a length-1 categorical is not a scalar and never has been (from a pandas point of view).

I'm pretty sure this special-case arrived in error at some point in the past.